server/src/main/java/org/elasticsearch/common/settings/AddStringKeyStoreCommand.java

```
@@ -56,44 +56,39 @@ InputStream getStdin() {
+            if (setting == null) {
+                throw new UserException(ExitCodes.USAGE, "The setting name can not be null");
+            }
+            if (keystore.getSettingNames().contains(setting) && options.has(forceOption) == false) {
+                if (terminal.promptYesNo("Setting " + setting + " already exists. Overwrite?", false) == false) {
+                    terminal.println("Exiting without modifying keystore.");
+                    return;
+                }
             }
-        }
 
-        final char[] value;
-        if (options.has(stdinOption)) {
-            BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
-            value = stdinReader.readLine().toCharArray();
-        } else {
-            value = terminal.readSecret("Enter value for " + setting + ": ");
-        }
+            final char[] value;
+            if (options.has(stdinOption)) {
+                BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
+                value = stdinReader.readLine().toCharArray();
+            } else {
+                value = terminal.readSecret("Enter value for " + setting + ": ");
+            }
 
-        try {
-            keystore.setString(setting, value);
-        } catch (IllegalArgumentException e) {
-            throw new UserException(ExitCodes.DATA_ERROR, "String value must contain only ASCII");
+            try {
+                keystore.setString(setting, value);
+            } catch (IllegalArgumentException e) {
+                throw new UserException(ExitCodes.DATA_ERROR, "String value must contain only ASCII");
+            }
+            keystore.save(env.configFile(), keyAndPass.getPassphrase());
+        } catch (SecurityException e) {
+            throw new UserException(ExitCodes.DATA_ERROR, "Failed to access the keystore. Please make sure the passphrase was correct.");
```

lets not swallow this exception; we should add them to the `UserException`